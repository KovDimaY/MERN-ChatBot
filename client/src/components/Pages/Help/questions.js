const homepage = {
  title: 'Home Page',
  questions: [
    'What is your name?',
    'What is your profession?',
    'Where do you live?',
    'How can I contact you?',
    'How old are you?',
    'Where are you from?',
    'Tell me something about yourself',
    'What languages do you speak?',
    'What are your hobbies?',
    'Tell me more about [programming, sports, puzzles, numismatics, photography]',
  ],
};

const experience = {
  title: 'Experience Page',
  questions: [
    'Tell me about your working experiences?',
    'What is your position in [Cornerjob, Bigfinite, Enixan, MBAS, Icicle]?',
    'When did you work in [Cornerjob, Bigfinite, Enixan, MBAS, Icicle]?',
    'Describe your work in [Cornerjob, Bigfinite, Enixan, MBAS, Icicle]?',
    'What technologies did you use in [Cornerjob, Bigfinite]?',
    'What responsibilities did you have in [Enixan, MBAS]?',
  ],
};

const education = {
  title: 'Education Page',
  questions: [
    'Tell me about your education?',
    'What topic did you study during you [master, bachelor]?',
    'When did you get your [master, bachelor]?',
    'What [roles, activities] did you have while getting your [master, bachelor]?',
    'What additional education do you have?',
  ],
};

const projects = {
  title: 'Projects Page',
  questions: [
    'Tell me about your projects?',
    'I want to know more about [TripReviewer, SimpleChat, Highcharts, Memory Game, KovDimaY]?',
    'Which technologies did you use in [TripReviewer, SimpleChat, Highcharts, Memory Game, KovDimaY]?',
  ],
};

export const questions = [homepage, experience, education, projects];
