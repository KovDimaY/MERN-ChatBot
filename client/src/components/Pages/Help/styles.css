.help-page-container {
  padding-bottom: 100px;
  max-width: 800px;
  margin: auto;
}

.help-page-container .text {
  font-size: 18px;
  padding: 0 15px;
  text-indent: 30px;
  text-align: justify;
}

.help-page-container .help-section-title,
.help-page-container .help-section-subtitle {
  text-align: center;
  margin: 35px 0;
}

.help-page-container .sections-wrapper {
  padding: 0 30px;
  list-style-type: initial;
}

.help-page-container .section-title {
  display: block;
  background: #eceff1;
  color: #424242;
  padding: 10px;
  border-radius: 2px;
  font-weight: bold;
  font-size: 16px;
}

.help-page-container .shortcut-links-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.help-page-container .shortcut-links-wrapper .btn-large {
  width: auto;
}

.help-page-container .questions-wrapper {
  margin-bottom: 15px;
}

.help-page-container .question-item {
  margin: 5px 0;
  padding: 5px 25px;
  color: #757575;
  border-bottom: 1px solid #eceff1;
}

.help-page-container .question-item:before {
  content: ' - ';
}

.help-page-container .question-item:last-child {
  border-bottom: none;
}

@media (max-width: 768px) {
  .help-page-container .shortcut-links-wrapper {
    flex-direction: column;
    text-align: center;
  }

  .help-page-container .shortcut-links-wrapper .btn-large {
    width: 250px;
    margin-bottom: 15px;
  }
}

@media (max-width: 600px) {
  .help-page-container .help-section-title {
    font-size: 32px;
    margin: 25px 0;
  }

  .help-page-container .help-section-subtitle {
    font-size: 24px;
    margin: 25px 0;
  }

  .help-page-container .text {
    font-size: 16px;
    padding: 0;
  }

  .help-page-container .sections-wrapper {
    padding: 0;
  }
}
