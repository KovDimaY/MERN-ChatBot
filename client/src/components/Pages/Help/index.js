import React from 'react';
import { Link } from 'react-router-dom';

import { STORYBOOK, STYLEGUIDE, NOT_FOUND } from 'constants/routes';
import { questions } from './questions';

import './styles.css';

const HelpPage = () => {
  const renderQuestions = items =>
    items.map(question => (
      <li key={question} className="question-item">
        {question}
      </li>
    ));

  const renderSections = sections =>
    sections.map(section => (
      <li key={section.title}>
        <span className="section-title">{section.title}</span>
        <ul className="questions-wrapper">
          {renderQuestions(section.questions)}
        </ul>
      </li>
    ));

  const renderLinks = () => (
    <ul className="shortcut-links-wrapper">
      <li>
        <a
          className="waves-effect waves-light btn-large light-blue lighten-1"
          href={STORYBOOK}
        >
          <i className="material-icons left">book</i>
          Storybook
        </a>
      </li>
      <li>
        <a
          className="waves-effect waves-light btn-large purple lighten-1"
          href={STYLEGUIDE}
        >
          <i className="material-icons left">widgets</i>
          Styleguidist
        </a>
      </li>
      <li>
        <Link
          className="waves-effect waves-light btn-large teal lighten-1"
          to={NOT_FOUND}
        >
          <i className="material-icons left">location_off</i>
          404 page
        </Link>
      </li>
    </ul>
  );

  return (
    <div className="help-page-container">
      <h3 className="help-section-title">Help</h3>
      <p className="text">
        As this is a learning project, I want you to access easily to all the
        hidden places of the project, even if normally users should not access
        them. In my case, they are: 404 page and documentations in Sorybook and
        Styleguidist. This is something internal but maybe someone of you will
        want to check these pages too, so here you have the links:
      </p>
      {renderLinks()}
      <p className="text">
        Also, if you are completely stuck and you have no idea what question to
        ask in order to open some data you want to get - here you can see the
        list of examples for all the questions that you need to open 100% of the
        information on all the pages. As you can see, it is pretty
        straightforward and simple to open all the data and you do not need to
        ask a lot.
      </p>
      <p className="text">
        Notice, that these questions are only examples and you do not have to
        ask exactly the same questions. My bot should understand many different
        ways of asking about these topics. So you are welcome try different
        variations to see how flexible it is. And if you are lazy and you do not
        want to play with the bot - here you can find a link to my{' '}
        <a
          href="https://www.linkedin.com/in/kovalenkodmytro/"
          target="_blank"
          rel="noopener noreferrer"
        >
          LinkedIn
        </a>{' '}
        profile and see everything at once.
      </p>
      <p className="text">
        Remember, <b>your feedback is very important</b> to me! If you see that
        my bot answers in a weird way to a simple question, or if you think that
        something can be improved in a visual, user experience or architectural
        way - please, reach me aut and share your opinion! I will appreciate a
        lot your time and effort to make my project better! Thank you in
        advance! :D
      </p>
      <h4 className="help-section-subtitle">Examples of questions:</h4>
      <ul className="sections-wrapper">{renderSections(questions)}</ul>
    </div>
  );
};

export default HelpPage;
