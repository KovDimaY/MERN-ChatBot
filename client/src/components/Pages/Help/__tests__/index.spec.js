import React from 'react';
import { create } from 'react-test-renderer';

import Help from '../../Help';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

const mockComponent = props => <Help {...props} />;

describe('components/Pages/<Help />', () => {
  it('should render component with no feedbacks', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
