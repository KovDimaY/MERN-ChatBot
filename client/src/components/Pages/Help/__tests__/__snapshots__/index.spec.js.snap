// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pages/<Help /> should render component with no feedbacks 1`] = `
<div
  className="help-page-container"
>
  <h3
    className="help-section-title"
  >
    Help
  </h3>
  <p
    className="text"
  >
    As this is a learning project, I want you to access easily to all the hidden places of the project, even if normally users should not access them. In my case, they are: 404 page and documentations in Sorybook and Styleguidist. This is something internal but maybe someone of you will want to check these pages too, so here you have the links:
  </p>
  <ul
    className="shortcut-links-wrapper"
  >
    <li>
      <a
        className="waves-effect waves-light btn-large light-blue lighten-1"
        href="/docs/storybook"
      >
        <i
          className="material-icons left"
        >
          book
        </i>
        Storybook
      </a>
    </li>
    <li>
      <a
        className="waves-effect waves-light btn-large purple lighten-1"
        href="/docs/styleguide"
      >
        <i
          className="material-icons left"
        >
          widgets
        </i>
        Styleguidist
      </a>
    </li>
    <li>
      <Link
        className="waves-effect waves-light btn-large teal lighten-1"
        to="/not-found"
      >
        <i
          className="material-icons left"
        >
          location_off
        </i>
        404 page
      </Link>
    </li>
  </ul>
  <p
    className="text"
  >
    Also, if you are completely stuck and you have no idea what question to ask in order to open some data you want to get - here you can see the list of examples for all the questions that you need to open 100% of the information on all the pages. As you can see, it is pretty straightforward and simple to open all the data and you do not need to ask a lot.
  </p>
  <p
    className="text"
  >
    Notice, that these questions are only examples and you do not have to ask exactly the same questions. My bot should understand many different ways of asking about these topics. So you are welcome try different variations to see how flexible it is. And if you are lazy and you do not want to play with the bot - here you can find a link to my
     
    <a
      href="https://www.linkedin.com/in/kovalenkodmytro/"
      rel="noopener noreferrer"
      target="_blank"
    >
      LinkedIn
    </a>
     
    profile and see everything at once.
  </p>
  <p
    className="text"
  >
    Remember, 
    <b>
      your feedback is very important
    </b>
     to me! If you see that my bot answers in a weird way to a simple question, or if you think that something can be improved in a visual, user experience or architectural way - please, reach me aut and share your opinion! I will appreciate a lot your time and effort to make my project better! Thank you in advance! :D
  </p>
  <h4
    className="help-section-subtitle"
  >
    Examples of questions:
  </h4>
  <ul
    className="sections-wrapper"
  >
    <li>
      <span
        className="section-title"
      >
        Home Page
      </span>
      <ul
        className="questions-wrapper"
      >
        <li
          className="question-item"
        >
          What is your name?
        </li>
        <li
          className="question-item"
        >
          What is your profession?
        </li>
        <li
          className="question-item"
        >
          Where do you live?
        </li>
        <li
          className="question-item"
        >
          How can I contact you?
        </li>
        <li
          className="question-item"
        >
          How old are you?
        </li>
        <li
          className="question-item"
        >
          Where are you from?
        </li>
        <li
          className="question-item"
        >
          Tell me something about yourself
        </li>
        <li
          className="question-item"
        >
          What languages do you speak?
        </li>
        <li
          className="question-item"
        >
          What are your hobbies?
        </li>
        <li
          className="question-item"
        >
          Tell me more about [programming, sports, puzzles, numismatics, photography]
        </li>
      </ul>
    </li>
    <li>
      <span
        className="section-title"
      >
        Experience Page
      </span>
      <ul
        className="questions-wrapper"
      >
        <li
          className="question-item"
        >
          Tell me about your working experiences?
        </li>
        <li
          className="question-item"
        >
          What is your position in [Cornerjob, Bigfinite, Enixan, MBAS, Icicle]?
        </li>
        <li
          className="question-item"
        >
          When did you work in [Cornerjob, Bigfinite, Enixan, MBAS, Icicle]?
        </li>
        <li
          className="question-item"
        >
          Describe your work in [Cornerjob, Bigfinite, Enixan, MBAS, Icicle]?
        </li>
        <li
          className="question-item"
        >
          What technologies did you use in [Cornerjob, Bigfinite]?
        </li>
        <li
          className="question-item"
        >
          What responsibilities did you have in [Enixan, MBAS]?
        </li>
      </ul>
    </li>
    <li>
      <span
        className="section-title"
      >
        Education Page
      </span>
      <ul
        className="questions-wrapper"
      >
        <li
          className="question-item"
        >
          Tell me about your education?
        </li>
        <li
          className="question-item"
        >
          What topic did you study during you [master, bachelor]?
        </li>
        <li
          className="question-item"
        >
          When did you get your [master, bachelor]?
        </li>
        <li
          className="question-item"
        >
          What [roles, activities] did you have while getting your [master, bachelor]?
        </li>
        <li
          className="question-item"
        >
          What additional education do you have?
        </li>
      </ul>
    </li>
    <li>
      <span
        className="section-title"
      >
        Projects Page
      </span>
      <ul
        className="questions-wrapper"
      >
        <li
          className="question-item"
        >
          Tell me about your projects?
        </li>
        <li
          className="question-item"
        >
          I want to know more about [TripReviewer, SimpleChat, Highcharts, Memory Game, KovDimaY]?
        </li>
        <li
          className="question-item"
        >
          Which technologies did you use in [TripReviewer, SimpleChat, Highcharts, Memory Game, KovDimaY]?
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
